homework-2/middewares/userMiddlewares.ts

```diff
@@ -27,26 +27,40 @@ export function getUserByIdHandler(req: Request<IIdParam>, res: Response): void
     }
 }
 
-export function getAutoSuggestHandler(req: Request<unknown, unknown, unknown, IAutoSuggestReqQuery>, res: Response): void {
-    const autoSuggests = UserService.getAutoSuggestUsers(req.query.loginSubstring, req.query.limit);
+export async function getAutoSuggestHandler(
+    req: Request<unknown, unknown, unknown, IAutoSuggestReqQuery>,
+    res: Response
+): Promise<void> {
+    const autoSuggests = await UserService.getAutoSuggestUsers(req.query.loginSubstring, req.query.limit);
     res.send(autoSuggests);
 }
 
-export function addUserHandler(req: Request<Record<string, unknown>, unknown, IUser>, res: Response): void {
+export async function addUserHandler(
+    req: Request<Record<string, unknown>, unknown, IUserAttrs>,
+    res: Response
+): Promise<void> {
     const errors = validationResult(req);
```

Comment:
I would use separate handler for validation

Reply:
todo for q&a session